dbt_platform_helper/providers/ecr.py

```diff
@@ -0,0 +1,20 @@
+from boto3 import Session
+
+from dbt_platform_helper.utils.aws import get_aws_session_or_abort
+
+
+class ECRProvider:
+    def __init__(self, session: Session = None):
+        self.session = session
+        self.client = None
+
+    def _get_client(self):
+        if not self.session:
+            self.session = get_aws_session_or_abort()
+        return self.session.client("ecr")
+
+    def get_ecr_repo_names(self):
+        return [
+            repo["repositoryName"]
+            for repo in self._get_client().describe_repositories().get("repositories", {})
```

Comment:
we may want to use the boto3 paginator for describe calls. Recently we have had a few bugs when results are >20

Reply:
I'll have a look.

Reply:
Example here: https://github.com/uktrade/platform-tools/blob/d9bbe4182e401a6d331d04ae929ab840dde82be0/dbt_platform_helper/utils/aws.py#L426